python/objToJSON.c

```diff
@@ -844,70 +882,90 @@ PyObject* objToJSON(PyObject* self, PyObject *args, PyObject *kwargs)
 
   PRINTMARK();
 
-  return newobj;
+  HPyTracker_Close(ctx, ht);
+  return h_ret;
 }
 
-PyObject* objToJSONFile(PyObject* self, PyObject *args, PyObject *kwargs)
+
+HPyDef_METH(objToJSONFile, "dump", objToJSONFile_impl, HPyFunc_KEYWORDS,
+            .doc="Converts arbitrary object recursively into JSON file. " \
+                 ENCODER_HELP_TEXT);
+static HPy
+objToJSONFile_impl(HPyContext ctx, HPy self, HPy *args, HPy_ssize_t nargs, HPy kw)
 {
-  PyObject *data;
-  PyObject *file;
-  PyObject *string;
+  HPy data;
+  HPy file;
+  PyObject *file_o;
+  HPy string;
+  PyObject *string_o;
   PyObject *write;
   PyObject *argtuple;
+  HPyTracker ht;
 
   PRINTMARK();
 
-  if (!PyArg_ParseTuple (args, "OO", &data, &file))
+  if (!HPyArg_Parse(ctx, &ht, args, nargs, "OO", &data, &file))
   {
-    return NULL;
+    return HPy_NULL;
   }
 
-  if (!PyObject_HasAttrString (file, "write"))
+  file_o = HPy_AsPyObject(ctx, file);
```

Comment:
Is it really necessary to convert to `PyObject *` here? I think we do already have all necessary API functions to operate on handles.